### DISCUSSION TITLE
React 19 support

### DISCUSSION THREAD

**Discussion by tnfAngel** [chakra-ui/chakra-ui#8839] [General]
### Description

Support for the new [React 19](https://react.dev/blog/2024/04/25/react-19), since many components do not work correctly.

### Problem Statement/Justification

For example, popovers are not placed correctly.

### Proposed Solution or API

Add a React 19 support branch
https://react.dev/blog/2024/04/25/react-19

**Comment by dmvjs** [comment]
popovers not being placed correctly could be resolved by wrapping them with a span, that's not a React 19 issue if so

**Comment by tnfAngel** [comment]
> popovers not being placed correctly could be resolved by wrapping them with a span, that's not a React 19 issue if so

![image](https://github.com/chakra-ui/chakra-ui/assets/57068341/3e71dcaa-1117-4d82-8ecd-ee3fdc528b71)
Did not solve the issue. 

I am going to assume that it is an issue caused by the react 19 update, since before updating it worked perfectly without any problem and after updating it stopped working (wrong position)

**Comment by oosawy** [comment]
it seems to work fine with minimum reproduction

https://stackblitz.com/edit/vitejs-vite-xm4mnb?file=src%2FApp.tsx

**Reply by futhr** [reply]
@oosawy FYI: Your StackBlitz example uses React 18.3.1

**Reply by yjb94** [reply]
I've forked your stackblitz and tried with React 19.0.0 and popover appeared in the right place.

**Comment by AKomar-sm** [comment]
After upgrading to React v19, I get an error when running `yarn build`
```
Attempted import error: 'createFactory' is not exported from 'react' (imported as 't').
```
"createFactory" is not used in my code. Do you have any suggestions?

**Reply by AKomar-sm** [reply]
Seems like not a React 19 issue. I updated Node.js and all packages and this one disappeared (in fact created it from scratch).

**Comment by Dammmien** [comment]
Hi, is it planned to support React 19 in Chakra UI V2 ?

**Comment by martinbroos** [comment]
When upgrading to React 19 i'm seeing no functional issues, but not all components are used in my project. So would be nice if we have an overview of what components would cause problems so a fix could be made. 

One problem I do have is that Chakra only allows a peer dependency of React 18 which causes CLI build to crash. Is it possible that we add React 19 to the allowed peer dependencies? Or it it to unstable to do so?

```
npm error Found: react@19.2.3
npm error node_modules/react
npm error   react@"^19.2.3" from the root project
npm error   peer react@">=18" from @chakra-ui/hooks@2.4.5
npm error   node_modules/@chakra-ui/hooks
npm error     @chakra-ui/hooks@"2.4.5" from @chakra-ui/react@2.10.9
npm error     node_modules/@chakra-ui/react
npm error       @chakra-ui/react@"^2.8.2" from the root project
npm error       1 more (@chakra-ui/next-js)
npm error   21 more (@chakra-ui/next-js, @chakra-ui/react, ...)
```

**Comment by sortizp** [comment]
Chakra UI not working with React 19 versions. Following Vite guide. I tried giving this a shot but is a dead-end. Setting it up with JavaScript.

**Comment by segunadebayo** [comment]
v3 supports React 19. If you are on v2, stay on 2.8.x. We are not doing a React 19 rewrite there.
